### ISSUE TITLE
interp/constrextern.ml Assertion failed when proving triple_factorec from Separation Logic Foundations

### ISSUE THREAD

**Issue by wikku** [coq/coq#19127]
### Description of the problem

[Rules.v: triple_factorec](https://softwarefoundations.cis.upenn.edu/slf-current/Rules.html#ExamplePrograms2.triple_factorec)

```
(* Rules.v in Separation Logic Foundations Version 2.1 (2024-01-03 14:54 *)

Lemma triple_factorec : forall n,
  n >= 0 ->
  triple (factorec n)
    \[]
    (fun r => \[r = facto n]).
Proof using.
intros.
induction n.
Focus 2.
applys* triple_app_fix.
simpl.

(*
Anomaly
"File "interp/constrextern.ml", line 214, characters 2-8: Assertion failed."
Please report at http://coq.inria.fr/bugs/.
*)
```

The error persists when using `eapply triple_app_fix. reflexivity.` or `applys triple_app_fix. reflexivity.`

### Version of Coq where this bug occurs

8.19.1

**Comment by SkySkimmer** [comment]
Looks like https://github.com/coq/coq/issues/18342
If it's the same bug then it should be fixed in master

**Comment by wikku** [comment]
Appears to be fixed on master (7ec2afdd). But I had to Admit broken proofs of `LibSet.v:{fold_isolate,card_disjoint_union_single}` to compile slf.
